README.md

```diff
@@ -513,33 +513,69 @@ public class ItemTag {
 }
 ``` 
 
-If you want to find all customers and fetch additional nested attributes (entities) to avoid `SELECT N+1 Problem` you can do the following:
+If you want to find all customers who ordered item with given tag name and fetch additional nested attributes (entities) to avoid `SELECT N+1 Problem` you can do the following:
 ```java
-@RequestMapping(value = "/findCustomersByOrderedItemTag")
+@RequestMapping(value = "/findCustomersWhoOrderedItemWithGivenTag")
 @PostMapping
 public Object findCustomers(
 	    @JoinFetch(paths = "orders", alias = "o")
-	    @JoinFetch(paths = "o.tags") Specification<Customer> spec) {
-	return customerRepo.findAll(spec, Sort.by("id"));
+	    @JoinFetch(paths = "o.tags", alias = "t")
+	    @Spec(path = "t.name", params = "tagName", spec = Equal.class) Specification<Customer> spec) {
+	return customerRepo.findAll(spec);
 }
 ```
 
 The same as in case of multi-level joins, annotations are processed sequentially, the order must be kept!
 
-__Join Fetch aliases exists only in context of join fetch and can't be used in another specs paths!__ 
-Following spec is invalid:
+  Please remember that:
+  * Join fetch path can use only aliases of another fetch joins. 
+  * Join path can use only aliases of another joins. 
 
-```java
-@RequestMapping(value = "/findCustomersByOrderedItemTag")
-@PostMapping
-public Object findCustomersByOrderedItemTag(
-		@JoinFetch(path = "orders", alias = "o")
-		@Spec(path = "o.itemName", params = "itemName", spec = Equal.class) Specification<Customer> spec) {
-	return customerRepo.findAll(spec, Sort.by("id"));
-}
-```
-
-If there is a need to refer to joined paths in other specs, then regular join (not fetch) should be used as described in the Join section.
+  Following combinations are forbidden:
```

Comment:
I think that we reached the level where README.md is very long. Maybe put these examples to a separate md file and just link here?

Reply:
(just an idea)